builtin/logical/consul/path_roles.go

```diff
@@ -64,13 +81,26 @@ func pathRolesWrite(
 
 	// Write the policy into storage
 	err = req.Storage.Put(&logical.StorageEntry{
-		Key:   "policy/" + d.Get("name").(string),
+		Key:   "policy/" + name,
 		Value: policyRaw,
```

Comment:
Can we instead just encode an object with the lease + policy instead of using two separate keys?